### PR TITLE
Sync with latest addon blueprint and use decorator-transforms

### PR DESCRIPTION
**Pull request by SergeAstapov**
v2 addon blueprint: https://github.com/embroider-build/addon-blueprint

Minor gain: using `decorator-transforms` reduced bundle size for the app.

### CODE REVIEW

**Comment by SergeAstapov** [comment]
side note: funnily enough, v5 scenarios pass only because repo uses yarn and yarn incorrectly handles peer dependencies

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -73,25 +67,24 @@ jobs:
...
+          - ember-lts-5.4
           - ember-release
           - ember-beta
           - ember-canary
-          - ember-default-with-jquery
-          - ember-classic
-#          - embroider-safe
-#          - embroider-optimized
+          # - embroider-safe
+          # - embroider-optimized
```
Comment (villander) [inline .github/workflows/ci.yml:77]: IMO does not make sense sync with embroider blueprint and don't support it on CI

File: README.md [hunk 2]
```diff
@@ -7,26 +7,20 @@ This addon provides an API for authoring a [Router service](https://api.emberjs.
...
 
 ## Compatibility
-------------------------------------------------------------------------------
-
-* Ember.js v3.12 or above
-* Ember CLI v3.24 or above
-* Node.js v14 or above
 
+- Ember.js v3.12 or above
+- Embroider or ember-auto-import v2
```
Comment (villander) [inline README.md:12]: Could you please link the embroider blueprint to the PR description for better testing and review @SergeAstapov ?

**Review by villander — CHANGES_REQUESTED** [review]
Couple requested changes! 

@SergeAstapov could you please describe what's size do you mean about the app? Are you talking about the testing-app?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (SergeAstapov) [reply .github/workflows/ci.yml:77]: @villander as this was commented out, I didn't touch. Let me open separate PR and uncomment these scenarios. I'm not sure why they been commented and hat was there error

**Comment by SergeAstapov** [comment]
> Couple requested changes!
> 
> @SergeAstapov could you please describe what's size do you mean about the app? Are you talking about the testing-app?

added link to PR description.

RE: `decorator-transforms/`
You may read about this package in https://github.com/ef4/decorator-transforms/
now it comes from v2 addon blueprint, see https://github.com/embroider-build/addon-blueprint/blob/main/files/__addonLocation__/package.json#L37

TL;DR `@babel/plugin-proposal-decorators` has large footprint in terms of compiled code size. In our app we saw 18% reduction in size for `vendor.js` and `app.js` once we introduced `decorator-transforms/` and disabled `@babel/plugin-proposal-decorators`.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (villander) [reply .github/workflows/ci.yml:77]: Oh sorry! nevermind

File: test-app/ember-cli-build.js [hunk 3]
```diff
@@ -4,20 +4,11 @@ const EmberApp = require('ember-cli/lib/broccoli/ember-app');
 
 module.exports = function (defaults) {
   const app = new EmberApp(defaults, {
-    // Add options here
+    autoImport: {
+      watchDependencies: ['ember-engines-router-service'],
```
Comment (villander) [inline test-app/ember-cli-build.js:8]: @SergeAstapov why do we need this config?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (villander) [reply .github/workflows/ci.yml:77]: @SergeAstapov have you tried enable them to make it compatible with embroider?

File: test-app/ember-cli-build.js (hunk 3 above)
Comment (SergeAstapov) [reply test-app/ember-cli-build.js:8]: @villander this tells `ember-auto-import` to pick up changes in the addon.

This makes `ember s` and `ember test --server` commands pick up changes in the `ember-engines-router-service` addon without restarting the command.

This is not unique to `ember-engines-router-service`, it's just a config now aded by blueprint https://github.com/embroider-build/addon-blueprint/blob/main/files/test-app-overrides/ember-cli-build.js#L9 and that was just not present in this repo before.

File: README.md (hunk 2 above)
Comment (SergeAstapov) [reply README.md:12]: @villander this was already done

File: .github/workflows/ci.yml (hunk 1 above)
Comment (SergeAstapov) [reply .github/workflows/ci.yml:77]: @villander no, it's out of scope of this PR.

File: test-app/ember-cli-build.js (hunk 3 above)
Comment (villander) [reply test-app/ember-cli-build.js:8]: @SergeAstapov could you please enable it only for non-production environments?

Comment (SergeAstapov) [reply test-app/ember-cli-build.js:8]: @villander this flag only affects `ember test --server` and `ember server` commands, hence this makes no effect for `ember build` command.

Comment (villander) [reply test-app/ember-cli-build.js:8]: Thank you @SergeAstapov

**Comment by villander** [comment]
We can merge this PR when CI get green again

**Comment by SergeAstapov** [comment]
@villander this is refreshed with the latest blueprint as of now and CI is passing

**Review by villander — APPROVED** [review]
Thank you! Sorry for the delay in merging it
